website/src/components/views/CreateJob.jsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import queryString from 'query-string'
+import { NavBar } from '../NavBar';
+
+class CreateJob extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onSubmitWithProps = this.onSubmitWithProps.bind(this)
+
+        const queryValues = queryString.parse(props.location.search)
+        let cronJob = {}
+        props.cronJobs.map(item => {
+            if (queryValues.namespace === item.namespace && queryValues.cronjob === item.name) {
+                cronJob = item
+            }
+            return null
+        })
+        let formValues = {}
+        cronJob.config.options.map((option, index) => {
+            formValues[index] = option.default
+            return null
+        })
+
+        this.state = {
+            errors: {},
+            queryValues: queryValues,
+            cronJob: cronJob,
+            formValues,
+        }
+    }
+
+    async onSubmitWithProps(event, props) {
+        event.preventDefault();
+        let jobRequest = new Array(props.cronJob.config.options.length)
+        let formValues = {}
+        props.cronJob.config.options.map((option, index) => {
+            jobRequest[index] = {
+                "envVar": option.envVar,
+                "container": option.container,
+                "value": event.target[index].value,
+            }
+            formValues[index] = event.target[index].value
+            return null
+        })
```

Comment:
Same thing here. Using `map` for populating an existing array is not the usual way.